src/server/svr_chk_owner.c

```diff
@@ -388,68 +389,93 @@ authenticate_user(struct batch_request *preq, struct connection *pcred)
  * 		chk_job_request - check legality of a request against a job
  * @par
  *		this checks the most conditions common to most job batch requests.
- *		It also returns a pointer to the job if found and the tests pass.
+ *		It returns TRUE if job id is valid and the tests pass else returns FALSE
+ *		It also returns job type and a pointer to the job if found.
  *		If the request is for a single subjob or a range of subjobs (of an
- *		Job Array),  the return job pointer is to the parent Array Job.
- * @par
- *		Depending on what the "jobid" identifies, the following is returned
- *		in the integer pointed to by rc:
- *	   	IS_ARRAY_NO (0)       - for a regular job
- *	   	IS_ARRAY_ArrayJob (1) - for an Array Job
- *	   	IS_ARRAY_Single (2)   - for a single subjob
- *	   	IS_ARRAY_Range (3)    - for a range of  subjobs
+ *		Job Array), also returns pointer to the parent Array Job and the offset of
+ *		the subjob in its parent array.
  *
  * @param[in]	jobid	-	Job Id.
  * @param[in,out]	preq	-	job batch request
- * @param[out]	rc	-	Depending on what the "jobid" identifies,
+ * @param[in,out]	jobtype	-	Depending on what the "jobid" identifies,
```

Comment:
Mention jobtype in this case as mandatory parameter. Apply the comment wherever required.